### PR TITLE
[SKY9X] Audio fix - Tested on 9XR-PRO

### PR DESCRIPTION
**Pull request by bsongis**

### CODE REVIEW

File: radio/src/audio_arm.h [hunk 1]
```diff
@@ -116,6 +118,7 @@ enum AudioBufferState
 struct AudioBuffer {
   audio_data_t data[AUDIO_BUFFER_SIZE];
   uint16_t size;
+  uint8_t  state;
```
Comment (projectkk2glider) [inline radio/src/audio_arm.h]: You could put `#if defined(PCBSKY9X)` around `state`.

Comment (bsongis) [inline radio/src/audio_arm.h]: indeed, I will also change the #ifdef and introduce AUDIO_DOUBLE_BUFFERING rather than a board type

Comment (bsongis) [inline radio/src/audio_arm.h]: as we have LCD_DUAL_BUFFER it will be AUDIO_DUAL_BUFFER

File: radio/src/audio_arm.h [hunk 2]
```diff
@@ -314,38 +317,79 @@ class AudioBufferFifo {
...
     void freeNextFilledBuffer()
     {
       audioDisableIrq();
+#if defined(PCBSKY9X)
+      if (audioBuffers[readIdx].state == AUDIO_BUFFER_PLAYING) {
+        audioBuffers[readIdx].state = AUDIO_BUFFER_FREE;
+        readIdx = nextBufferIdx(readIdx);
+        bufferFull = false;
+      }
+#else
```
Comment (bsongis) [inline radio/src/audio_arm.h]: I only ported the code from 'master' as I knew it works. I will try now to remove this code which gives me headaches

**Pull request merged**